editor/spriteFieldEditor.ts

```diff
@@ -0,0 +1,223 @@
+            if (this.preview) {
+                this.size_.height = this.preview.outerHeight() + 20;
+                this.size_.width = this.preview.outerWidth() + 20;
+            }
+        }
+
+        getText() {
+            if (!this.state) {
+                return "img``";
+            }
+            let res = "img`";
+
+            for (let r = 0; r < this.state.height; r++) {
+                res += "\n"
+                for (let c = 0; c < this.state.width; c++) {
+                    res += hexChars[this.state.get(c, r)] + " ";
+                }
+            }
+
+            res += "\n`";
+
+            return res;
+        }
+
+        setText(newText: string) {
+            if (newText == null) {
+                return;
+            }
+            this.parseBitmap(newText);
+
+            if (this.preview) {
+                this.redrawPreview();
+            }
+
+            super.setText(newText);
+        }
+
+        private redrawPreview() {
+            this.fieldGroup_.innerHTML = "";
+            this.preview = new mkcd.BitmapImage({
+                cellWidth: Math.floor(PREVIEW_WIDTH / 16),
+                cellHeight: Math.floor(PREVIEW_HEIGHT / 16),
+                cellClass: "pixel-cell"
+            }, this.state, [
+                    "rgba(0, 0, 0, 0)",
+                    "#000000",
+                    "#33e2e4",
+                    "#05b3e0",
+                    "#3d30ad",
+                    "#b09eff",
+                    "#5df51f",
+                    "#6a8927",
+                    "#65471f",
+                    "#98294a",
+                    "#f80000",
+                    "#e30ec0",
+                    "#ff9da5",
+                    "#ff9005",
+                    "#efe204",
+                    "#ffffff",
+                ]);
+
+            this.fieldGroup_.appendChild(this.preview.getView().el);
+        }
+
+        private parseBitmap(newText: string) {
+            // Strip the tagged template string business and the whitespace. We don't have to exhaustively
+            // replace encoded characters because the compiler will catch any disallowed characters and throw
+            // an error before the decompilation happens. 96 is backtick and 9 is tab
+            newText = newText.replace(/[ `]|(?:&#96;)|(?:&#9;)|(?:img)/g, "").trim();
+            newText = newText.replace(/&#10;/g, "\n");
+
+            const rows = newText.split("\n");
+
+            // We support "ragged" sprites so not all rows will be the same length
+            const sprite: number[][] = [];
+            let spriteWidth = 0;
+
+            for (let r = 0; r < rows.length; r++) {
+                const row = rows[r];
+                const rowValues: number[] = [];
+                for (let c = 0; c < row.length; c++) {
+                    // This list comes from libs/screen/targetOverrides.ts
+                    switch (row[c]) {
+                        case "0": case ".": rowValues.push(0); break;
```

Comment:
This is really dynamic, based on annotation on the `img()`

Reply:
Yeah, I'll make it configurable later. Should be a field option.